### PR TITLE
[new] Added RiTa@1.1.40 w/ git auto update

### PR DESCRIPTION
**Pull request by CyrusSUEN**
Referring to ticket https://github.com/cdnjs/cdnjs/issues/8442. Thank you so much for your hard work here at cdnjs!

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/dhowe/RiTaJS 
 * Official website (optional, not the repository): http://rednoise.org/rita/
 * NPM package url (optional): https://www.npmjs.com/package/rita
 * GitHub / Bitbucket popularity (required): 
   - Count of stars: 35
   - Count of watchers: 6
   - Count of forks: 10
 * NPM download stats (optional):
   - Downloads in the last day: 1
   - Downloads in the last week: 123
   - Downloads in the last month: 662

# Essential checklist
 * [ ] I'm the author of this library
   * [x] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [ ] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by PeterDaveHello** [comment]
@cdnjs/intern2 please review this PR.

File: ajax/libs/rita/package.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    "natural language",
+    "language",
+    "writing",
+    "generative"
+  ],
+  "license": "GPL-3.0",
+  "tonicExampleFilename": "tonic.js",
+  "optionalDependencies": {
+    "yamljs": "^0.2.3"
+  },
```
Comment (pvnr0082t) [inline ajax/libs/rita/package.json]: @CyrusSUEN Let's remove `tonicExampleFilename`, `optionalDependencies` fields.

File: ajax/libs/rita/package.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{
+  "name": "rita",
+  "version": "1.1.40",
+  "author": "Daniel C. Howe <daniel@rednoise.org>",
+  "description": "RiTa: A toolkit for natural language and computational literature",
+  "homepage": "http://rednoise.org/rita",
```
Comment (pvnr0082t) [inline ajax/libs/rita/package.json]: I think we can use `https` instead of `http` here.

File: ajax/libs/rita/package.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  "npmName": "rita",
+  "npmFileMap": [
+    {
+      "basePath": "/dist",
+      "files": [
+        "**/*"
+      ]
+    }
+  ],
+  "filename": "/rita-full.min.js"
```
Comment (pvnr0082t) [inline ajax/libs/rita/package.json]: I think the main file should be `"rita.min.js"`

File: ajax/libs/rita/package.json [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  ],
+  "license": "GPL-3.0",
+  "tonicExampleFilename": "tonic.js",
+  "optionalDependencies": {
+    "yamljs": "^0.2.3"
+  },
+  "npmName": "rita",
+  "npmFileMap": [
+    {
+      "basePath": "/dist",
```
Comment (pvnr0082t) [inline ajax/libs/rita/package.json]: There is no `dist` folder in npm package. I think we should use git auto-update here but I'm not sure. @PeterDaveHello Please give us some suggestions, thank you.

**Comment by pvnr0082t** [comment]
@CyrusSUEN Please change the permission of files to inexecutable, thank you.

File: ajax/libs/rita/package.json (hunk 4 above)
Comment (PeterDaveHello) [inline ajax/libs/rita/package.json]: I didn't see `dist` in rita@1.1.391.

File: ajax/libs/rita/package.json (hunk 2 above)
Comment (CyrusSUEN) [inline ajax/libs/rita/package.json]: Thank you sir! This is a nice catch!

**Comment by CyrusSUEN** [comment]
Thank you so much @pvnr0082t and @PeterDaveHello !
I've updated and pushed the changes. 

Please note that I've also changed the `url` attribute of `repository`  from https://github.com/dhowe/RiTa.git which is the Java version to https://github.com/dhowe/RiTaJS.git the JavaScript version.

**Comment by PeterDaveHello** [comment]
@CyrusSUEN Thanks, we'll review your changes later once you passed the test.

**Comment by PeterDaveHello** [comment]
@cdnjs/intern2 please review it.

**Comment by CyrusSUEN** [comment]
Since only `rita.js` file is present in [npm package](https://www.npmjs.com/package/rita) while 4 dist files available in the [RiTaJS Github repo](https://github.com/dhowe/RiTaJS.git). 

Do you think I should configure it to git auto update?
If so, am I supposed to replace `npmFileMap` attribute in package.json with the following? Thanks so much again!
```json
  "autoupdate": {
    "source": "git",
    "target": "https://github.com/dhowe/ritajs",
    "basePath": "dist",
    "files": [
      "**/*"
    ]
  }
```

**Comment by nolsherry** [comment]
@CyrusSUEN Could you help to add an [issue](https://github.com/dhowe/RiTaJS/issues) on github to report the problem about the difference between the files on npm and the files on github?

File: ajax/libs/rita/package.json [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  "npmName": "rita",
+  "npmFileMap": [
+    {
+      "basePath": "lib",
+      "files": [
+        "**/*"
+      ]
+    }
+  ],
+  "filename": "/rita.min.js"
```
Comment (pvnr0082t) [inline ajax/libs/rita/package.json]: Please remove the slash here.

Comment (PeterDaveHello) [inline ajax/libs/rita/package.json]: @pvnr0082t you should at least modify our tests and the already exist libs to reflect this requirement.

Comment (CyrusSUEN) [inline ajax/libs/rita/package.json]: > remove the slash here

Thanks! Slash removed.

> Please remove the slash here.

Thanks! Slash removed.

**Comment by CyrusSUEN** [comment]
> report the problem about the difference between the files on npm and the files on github

I've notified the author. Not sure if that was an intended decision to put just one file on the npm package as I haven't heard back from him.
If that was the case, I will be sure to update the `package.json` to do git auto update instead.

**Comment by CyrusSUEN** [comment]
The author wants to keep the download size of the npm package as small as possible. Changed to git auto update. Please review

File: ajax/libs/rita/package.json [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  "license": "GPL-3.0",
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/dhowe/RiTaJS.git",
+    "basePath": "dist",
+    "files": [
+      "rita-full.js",
+      "rita-full.min.js",
+      "rita.js",
+      "rita.min.js"
```
Comment (PeterDaveHello) [inline ajax/libs/rita/package.json]: Let's use single `"**/*"` to grab all the files here.

Comment (CyrusSUEN) [inline ajax/libs/rita/package.json]: > single "**/*"

Thank you! Noted and updated.

**Pull request merged**